Framework/API/src/Algorithm.cpp

```diff
@@ -1616,6 +1622,168 @@ void Algorithm::setAlgStartupLogging(const bool enabled) {
 bool Algorithm::getAlgStartupLogging() const {
   return m_isAlgStartupLoggingEnabled;
 }
+
+/// Runs the algorithm with the specified execution mode.
+void Algorithm::exec(Parallel::ExecutionMode executionMode) {
+  switch (executionMode) {
+  case Parallel::ExecutionMode::Serial:
+  case Parallel::ExecutionMode::Identical:
+    return exec();
+  case Parallel::ExecutionMode::Distributed:
+    return execDistributed();
+  case Parallel::ExecutionMode::MasterOnly:
+    return execMasterOnly();
+  default:
+    throw(std::runtime_error("Algorithm " + name() +
+                             " does not support execution mode " +
+                             Parallel::toString(executionMode)));
+  }
```

Comment:
@martyngigg If the storage mode of the workspace is `StorageMode::Cloned` the corresponding execution mode would usually be `ExecutionMode::Identical`. You could for example imagine loading a monitor spectrum into a cloned workspace and then use it normalize a distributed workspace (I am not certain we will actually handle monitors in this case, but there are also other use cases, I think).